### ISSUE TITLE
[Mechanic] Player should remain in combat if your pet is still in combat.

### ISSUE THREAD

**Issue by SuddenDeath** [SunstriderEmu/BugTracker#1707]
**Describe the bug**
Currently if you get in combat, and your pet stays in combat.. You will still lose combat after 6 seconds even though your pet is still in combat.

**To Reproduce**
1. Get in combat with a player, put pet on that player
2. Stop doing any actions
3. You should never lose combat as long as pet is still in combat, currently you do.

**Expected behavior**
You should never lose combat if your pet is still in combat, if you've also been in combat during the same combat session as your pet.

TLDR: 
Pets shouldn't cause combat but if you're already in combat they keep you in it.

**Proof:**
https://youtu.be/D16d5kzadYQ?t=1m39s

See discussion here: https://www.warmane.com/bugtracker/report/44663

**Comment by z5n** [comment]
seems like current behavior is that pets DO keep you in combat, however they also cause you to enter combat through their actions
